spec/default_spec.rb

```diff
@@ -9,6 +9,40 @@
       it 'converges successfully' do
         expect { chef_run }.to_not raise_error
       end
+      it do
+        expect(chef_run.node['varnish']['version']).to eq(3)
+      end
+      it do
+        expect(chef_run.node['varnish']['log_daemon']).to eq(false)
+      end
+      it do
+        expect(chef_run).to include_recipe('varnish')
+      end
+      it do
+        expect(chef_run).to create_template('/etc/sysconfig/varnishncsa').with(
+          source: 'varnishncsa.erb',
+          variables: { directives: chef_run.node['osl-varnish']['ncsa_directives'] }
+        )
+      end
+      it do
+        expect(chef_run).to render_file('/etc/sysconfig/varnishncsa').with_content(
+          '# Generated by Chef for fauxhai.local
+#
+# Do not edit this file locally. Changes will be overwritten.'
+        )
```

Comment:
I wouldn't check the comments that are generated, just the content you generate. From what I said above, test the output of the attribute is being generated correctly.